plugins/background_fetch/background_fetch.cc

```diff
@@ -608,23 +619,49 @@ TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf */, int /
 {
   TSCont cont           = TSContCreate(cont_handle_response, nullptr);
   BgFetchConfig *config = new BgFetchConfig(cont);
-
-  // Parse the optional rules, wihch becomes a linked list of BgFetchRule's.
-  if (argc > 2) {
-    TSDebug(PLUGIN_NAME, "config file %s", argv[2]);
-    config->readConfig(argv[2]);
+  bool success          = true;
+
+  // The first two arguments are the "from" and "to" URL string. We need to
+  // skip them, but we also require that there be an option to masquerade as
+  // argv[0], so we increment the argument indexes by 1 rather than by 2.
+  argc--;
+  argv++;
+
+  // This is for backwards compatibility, ugly! ToDo: Remove for ATS v9.0.0 IMO.
+  if (argc > 1 && *argv[1] != '-') {
+    TSDebug(PLUGIN_NAME, "config file %s", argv[1]);
+    if (!config->readConfig(argv[1])) {
+      success = false;
+    }
+  } else {
+    if (config->parseOptions(argc, const_cast<const char **>(argv))) {
+      // Create the global log file. Note that calling this multiple times currently has no
+      // effect, only one log file is ever created.
+      if (config->logFile().size()) {
+        BgFetchState::getInstance().createLog(config->logFile());
+      }
+    } else {
+      success = false;
+    }
   }
 
-  *ih = (void *)config;
+  if (success) {
+    *ih = (void *)config;
```

Comment:
NO C STYLE CASTS!

Reply:
I'm actually surprised a cast is needed, as a pointer of any type should auto convert to a `void*`.

Reply:
I don’t remember, the code was like that before :)